system/scripts/zone/items/exp_cards.cs

```diff
@@ -53,4 +57,91 @@ public DialogTxResult MULTIPLE_USE_XPCARD(Character character, DialogTxArgs args
 
 		return DialogTxResult.Okay;
 	}
+
+	[ScriptableFunction]
+	public static DialogTxResult SCR_ITEM_EXP_UP(Character character, DialogTxArgs args)
+	{
+		if (args.TxItems.Length < 2)
+		{
+			Send.ZC_ADDON_MSG(character, AddonMessage.ITEM_EXP_STOP);
+			return DialogTxResult.Fail;
+		}
+
+		if (!character.IsSitting)
+		{
+			character.SystemMessage("AvailableOnlyWhileResting");
+			return DialogTxResult.Fail;
+		}
+
+		var targetItem = args.TxItems[0].Item;
+		var tgtGroup = targetItem.Data.Group;
+		var tgtEquipXpGroup = targetItem.Data.EquipExpGroup;
+
+		if (tgtEquipXpGroup == EquipExpGroup.None)
+		{
+			Send.ZC_ADDON_MSG(character, AddonMessage.ITEM_EXP_STOP);
+			return DialogTxResult.Fail;
+		}
+
+		foreach (var txItem in args.TxItems)
+		{
+			if (txItem.Item.IsExpired)
+			{
+				character.SystemMessage("CannotUseLifeTimeOverItem");
+				return DialogTxResult.Fail;
+			}
+		}
+
+		var synthesisTime = 5;
+		Send.ZC_ADDON_MSG(character, AddonMessage.ITEM_EXP_START, synthesisTime, "");
+		Send.ZC_CANCEL_MOUSE_MOVE(character);
+
+		if (!character.Components.TryGet<TimeActionComponent>(out var timeAction))
+			return DialogTxResult.Fail;
+
+		var timeActionResult = timeAction.StartAsync("ItemCraftProcess", "None", "UPGRADEGEM", TimeSpan.FromSeconds(synthesisTime)).Result;
```

Comment:
Using `Result` on the task isn't ideal, because by executing this synchronously, you're putting the thread that is handling this request that came in via the network on pause until the time action returns. Granted, it's only a couple seconds, but we want to handle packets as quickly as possible.

I would recommend using `Start` and a callback instead, or create an additional `async Task` method that you call with `CallSafe` from this one,  to more properly implement this asynchronous operation.